ui/src/main/resources/Meeting/Code/WebHomeSheet.xml

```diff
@@ -36,41 +36,92 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content>{{include reference="Meeting.Code.Migrations.MigrationMacros"/}}
+  <content>{{include reference="Licenses.Code.VelocityMacros"/}}
+{{include reference="Meeting.Code.Migrations.MigrationMacros"/}}
 
 {{velocity}}
-#set($docextras = [])
-#set ($discard = $xwiki.ssfx.use('uicomponents/widgets/userpicker/userPicker.css'))
-#set ($discard = $xwiki.ssfx.use('uicomponents/pagination/pagination.css', 'true'))
-$services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&gt;&gt;Meeting.CalendarView]]'])
-#set ($columnsProperties = {
-  'doc.title': {'type': 'text', 'size': 10, 'link': 'view'},
-  'leader': {'type': 'text', 'size': 10, 'html': true},
-  'place': {'type': 'text', 'size': 10},
-  'startDate': {'type': 'text', 'size': 10},
-  'endDate': {'type': 'hidden'},
-  'durationDisplay': {'type': 'text', 'filterable': true, 'sortable': false, 'html': true},
-  'status': {'type': 'list', 'size': 10},
-  '_actions': {'sortable': false, 'filterable': false, 'html': true, 'actions': ['delete', 'edit']}
-})
-#set ($options = {
-  'className': 'Meeting.Code.MeetingClass',
-  'translationPrefix': 'meeting.livetable.',
-  'tagCloud': true,
-  'rowCount': 15,
-  'selectedColumn': 'startDate'
-})
-#if ($request.getParameter('meeting') == 'all')
-  $services.localization.render('contrib.meeting.home.allMeetingsMessage', ['[[', "&gt;&gt;${doc}]]"])
-  #set ($options.defaultOrder = 'desc')
+#set ($configClassReference = $services.model.createDocumentReference('', ['Meeting', 'Code'], 'ConfigurationClass'))
+#set ($hasLicense = $services.licensing.licensor.hasLicensureForEntity($configClassReference))
+#if (!$hasLicense)
+  {{error}}#getMissingLicenseMessage('meeting.extension.name'){{/error}}
 #else
-  $services.localization.render('contrib.meeting.home.initialMessage', ['[[**', "&gt;&gt;||queryString='meeting=all']]"])
+## Display tabs
+#if("$!{view}" == '')
+  #set($view = $request.getParameter('view'))
+  #if("$!{view}" == '')
+    #set ($configClass = "Meeting.Code.ConfigurationClass")
+    #set ($configDoc = $xwiki.getDocument('Meeting.Code.Configuration'))
+    #set ($configObj = $configDoc.getObject($configClass))
+    #set ($view = $configObj.getProperty('defaultTab').getValue())
+    #if("$!{view}" == '')
+      #set ($view = 'past')
+    #end
+  #end
+#end
+
+#set($tabs = [])
+#macro(addAllDocsTab $tab)
+  #if($xwiki.hasAccessLevel('view', "$!xcontext.user", $tab.get('document')))
```

Comment:
Where is this 'document' key set? I don't see it when the macro is called below. Also, it would be better to use the security script service, see https://extensions.xwiki.org/xwiki/bin/view/Extension/Security+Module#HScriptingAPI28since6.1RC129 .

Reply:
Initially I wanted to have different documents for each tab, thus the 'document' key. I gave up on the idea when I later found out that there would be a lot of code duplication between the 'Past Meetings' and 'Future Meetings' tabs. Since the key won't be needed anymore, neither is the macro "maybeAddMeetingsTab", so I'll refactor the code accordingly.